query/src/main/scala/hmda/query/projections/institutions/InstitutionView.scala

```diff
@@ -0,0 +1,101 @@
+package hmda.query.projections.institutions
+
+import akka.actor.{ ActorRef, ActorSystem, Props }
+import akka.persistence.{ RecoveryCompleted, SnapshotOffer }
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.Sink
+import hmda.model.institution.Agency.CFPB
+import hmda.model.institution.InstitutionType.Bank
+import hmda.model.institution.{ Inactive, Institution }
+import hmda.persistence.messages.CommonMessages.{ Command, Event, GetState, Shutdown }
+import hmda.persistence.messages.events.institutions.InstitutionEvents._
+import hmda.persistence.model.HmdaPersistentActor
+import hmda.persistence.processing.HmdaQuery._
+
+object InstitutionView {
+
+  val name = "institutions-view"
+
+  case class GetInstitutionById(institutionId: String) extends Command
+  case class GetInstitutionsById(ids: List[String]) extends Command
+  case class LastProcessedEventOffset(seqNr: Long)
+  case object StreamCompleted
+
+  def props(): Props = Props(new InstitutionView)
+
+  def createInstitutionQuery(system: ActorSystem): ActorRef = {
+    system.actorOf(InstitutionView.props(), "institutions-view")
+  }
+
+  case class InstitutionViewState(institutions: Set[Institution] = Set.empty[Institution], seqNr: Long = 0L) {
+    def updated(event: Event): InstitutionViewState = {
+      event match {
+        case InstitutionCreated(i) =>
+          InstitutionViewState(institutions + i, seqNr)
+        case InstitutionModified(i) =>
+          val others = institutions.filterNot(_.id == i.id)
+          InstitutionViewState(others + i, seqNr)
+      }
+    }
+  }
+
+}
+
+class InstitutionView extends HmdaPersistentActor {
```

Comment:
To check my knowledge, I believe that in the future this actor will a different backend from `InstitutionPersistence` for storing data. This one will be optimized for reads, whereas the one for `InstitutionPersistence` will be optimized for writes. Is that correct?